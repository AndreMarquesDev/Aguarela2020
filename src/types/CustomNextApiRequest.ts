import type { NextApiRequest } from 'next';
import type { FormPostRequestBody } from './FormPostRequestBody';

export type CustomNextApiRequest = {
    body: FormPostRequestBody;
} & NextApiRequest;
