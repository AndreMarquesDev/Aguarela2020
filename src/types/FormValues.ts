export type FormValues = {
    name: string;
    email: string;
    message: string;
    brand: string;
    subject: string;
};
