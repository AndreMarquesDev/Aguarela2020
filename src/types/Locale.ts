export enum Locale {
    Pt = 'pt',
    En = 'en',
}
