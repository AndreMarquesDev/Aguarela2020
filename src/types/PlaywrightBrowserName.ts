export enum PlaywrightBrowserName {
    Chromium = 'chromium',
    Firefox = 'firefox',
    Webkit = 'webkit',
}
