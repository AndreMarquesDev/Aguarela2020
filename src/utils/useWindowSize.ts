import { useEffect, useState } from 'react';

type WindowSize = {
    width: number;
    height: number;
};

export const useWindowSize = (): WindowSize => {
    const [windowSize, setWindowSize] = useState<WindowSize>({
        width: 0,
        height: 0,
    });

    useEffect(() => {
        const handleResize = (): void => {
            // eslint-disable-next-line react-hooks-extra/no-direct-set-state-in-use-effect
            setWindowSize({
                width: window.innerWidth,
                height: window.innerHeight,
            });
        };

        window.addEventListener('resize', handleResize);

        handleResize();

        return (): void => window.removeEventListener('resize', handleResize);
    }, []);

    return windowSize;
};

export enum Breakpoint {
    Desktop = 1280,
    Phablet = 1023,
    Tablet = 767,
    Mobile = 511,
}
