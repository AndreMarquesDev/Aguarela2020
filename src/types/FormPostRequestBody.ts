import { FormValues } from './FormValues';

export interface FormPostRequestBody extends FormValues {
    isPlaywright: boolean;
}
