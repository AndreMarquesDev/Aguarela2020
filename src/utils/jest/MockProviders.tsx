/* eslint-disable react/jsx-no-constructed-context-values */
import React, { ReactNode } from 'react';
import { NavLinksContext } from '../../components/context/NavLinksContext';
import { TextsContext } from '../../components/context/TextsContext';
import { Locale } from '../../types/Locale';
import { textsEn, textsPt } from '../texts';

interface MockProvidersProps {
    language?: Locale;
    isMenuOpen?: boolean;
    toggleMenu?: jest.Mock;
    setNavHeight?: jest.Mock;
    children: ReactNode;
}

export const MockProviders = ({
    language = Locale.Pt,
    isMenuOpen = false,
    toggleMenu = jest.fn(),
    setNavHeight = undefined,
    children,
}: MockProvidersProps): JSX.Element => {
    return (
        <TextsContext.Provider
            value={{
                texts: language === Locale.Pt ? textsPt : textsEn,
            }}
        >
            <NavLinksContext.Provider value={{ isMenuOpen, toggleMenu, setNavHeight }}>
                {children}
            </NavLinksContext.Provider>
        </TextsContext.Provider>
    );
};
