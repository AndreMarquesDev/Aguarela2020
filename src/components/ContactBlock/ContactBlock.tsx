import type { Locale } from '../../types/Locale';
import { useRouter } from 'next/router';
import React, { useContext } from 'react';
import { contactBlockSectionDataTestId } from '../../utils/dataTestIds';
import { TextsContext } from '../context/TextsContext';
import { TextShadowBlock } from '../TextShadowBlock/TextShadowBlock';

export const ContactBlock = (): JSX.Element => {
    const { texts } = useContext(TextsContext);

    const { query } = useRouter();
    const currentLanguage = query.language?.toString() as Locale;
    const titleMobile
        = currentLanguage === 'en'
            ? `${texts.contactMe1}${texts.contactMe2}t`
            : `${texts.contactMe1}${texts.contactMe2}${texts.contactMe3}`;

    return (
        <TextShadowBlock
            hasButton
            dataTestId={contactBlockSectionDataTestId}
            textBlock1={texts.needHelpWithYourBusiness}
            textBlock2={texts.sendMeAnEmail}
            textBlock3={texts.iAmAvailableToAdvise}
            title1={texts.contactMe1}
            title2={texts.contactMe2}
            title3={texts.contactMe3}
            titleMobile={titleMobile}
        />
    );
};
