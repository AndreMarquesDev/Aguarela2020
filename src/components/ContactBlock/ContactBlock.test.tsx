import type { RenderResult } from '@testing-library/react';
import { render, screen } from '@testing-library/react';
import mockRouter from 'next-router-mock';
import React from 'react';
import { Locale } from '../../types/Locale';
import { MockProviders } from '../../utils/jest/MockProviders';
import { setJestWindowWidth } from '../../utils/jest/setJestWindowWidth';
import { textsEn, textsPt } from '../../utils/texts';
import { Breakpoint } from '../../utils/useWindowSize';
import { ContactBlock } from './ContactBlock';
import '@testing-library/jest-dom';

const renderComponent = (language: Locale = Locale.Pt): RenderResult => {
    return render(
        <MockProviders language={language}>
            <ContactBlock />
        </MockProviders>,
    );
};

describe('<ContactBlock />', () => {
    test('renders properly', () => {
        const { container } = renderComponent();

        expect(screen.getByText(textsPt.contactMe1)).toBeInTheDocument();
        expect(screen.getByText(textsPt.contactMe2)).toBeInTheDocument();
        expect(screen.getByText(textsPt.contactMe3)).toBeInTheDocument();
        expect(screen.getByText(textsPt.needHelpWithYourBusiness)).toBeInTheDocument();
        expect(screen.getByText(textsPt.sendMeAnEmail)).toBeInTheDocument();
        expect(screen.getByText(textsPt.iAmAvailableToAdvise)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly on mobile', () => {
        setJestWindowWidth(Breakpoint.Mobile);

        const { container } = renderComponent();

        expect(screen.queryByText(textsPt.contactMe1)).not.toBeInTheDocument();
        expect(screen.queryByText(textsPt.contactMe2)).not.toBeInTheDocument();
        expect(screen.queryByText(textsPt.contactMe3)).not.toBeInTheDocument();
        expect(
            screen.getByText(`${textsPt.contactMe1}${textsPt.contactMe2}${textsPt.contactMe3}`),
        ).toBeInTheDocument();
        expect(screen.getByText(textsPt.needHelpWithYourBusiness)).toBeInTheDocument();
        expect(screen.getByText(textsPt.sendMeAnEmail)).toBeInTheDocument();
        expect(screen.getByText(textsPt.iAmAvailableToAdvise)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly in English', () => {
        setJestWindowWidth(Breakpoint.Desktop);

        const { container } = renderComponent(Locale.En);

        expect(screen.getByText(textsEn.contactMe1)).toBeInTheDocument();
        expect(screen.getByText(textsEn.contactMe2)).toBeInTheDocument();
        expect(screen.getByText(textsEn.contactMe3)).toBeInTheDocument();
        expect(screen.getByText(textsEn.needHelpWithYourBusiness)).toBeInTheDocument();
        expect(screen.getByText(textsEn.sendMeAnEmail)).toBeInTheDocument();
        expect(screen.getByText(textsEn.iAmAvailableToAdvise)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly in English and mobile', () => {
        mockRouter.query = {
            language: 'en',
        };

        setJestWindowWidth(Breakpoint.Mobile);

        const { container } = renderComponent(Locale.En);

        expect(screen.queryByText(textsEn.contactMe1)).not.toBeInTheDocument();
        expect(screen.queryByText(textsEn.contactMe2)).not.toBeInTheDocument();
        expect(screen.queryByText(textsEn.contactMe3)).not.toBeInTheDocument();
        expect(screen.getByText(`${textsEn.contactMe1}${textsEn.contactMe2}t`)).toBeInTheDocument();
        expect(screen.getByText(textsEn.needHelpWithYourBusiness)).toBeInTheDocument();
        expect(screen.getByText(textsEn.sendMeAnEmail)).toBeInTheDocument();
        expect(screen.getByText(textsEn.iAmAvailableToAdvise)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
