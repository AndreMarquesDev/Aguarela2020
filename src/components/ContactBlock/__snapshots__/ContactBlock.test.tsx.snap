// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactBlock /> renders properly 1`] = `
<div>
  <section
    class="jsx-590009551 container"
    data-testid="contactBlock_section"
  >
    <div
      class="jsx-590009551 wrapper genericMargins"
    >
      <div
        class="jsx-590009551 titleBlock"
      >
        <strong
          class="jsx-590009551"
        >
          Con
        </strong>
        <strong
          class="jsx-590009551"
        >
          Ver
        </strong>
        <strong
          class="jsx-590009551"
        >
          Sar
        </strong>
      </div>
      <div
        class="jsx-590009551 textBlock"
      >
        <p
          class="jsx-590009551 bodyText bodyText--white"
          data-testid="textShadowBlock_textBlock1"
        >
          Precisa de ajuda com o seu negócio, projeto ou marca?
        </p>
        <p
          class="jsx-590009551 bodyText bodyText--white"
        >
          Envie-me um email e vamos conversar. Podemos começar o quanto antes a definir uma estratégia para social media de acordo com os seus objetivos.
        </p>
        <p
          class="jsx-590009551 bodyText bodyText--white"
        >
          Estou ainda disponível para dar consultoria em social media e partilhar todos os passos e dicas de que precisa para cumprir os seus objetivos.
        </p>
        <div
          class="jsx-2566809609 buttonWrapper"
        >
          <div
            class="jsx-2566809609 buttonBackground"
          >
            <a
              class="jsx-2566809609 button uppercase"
              href="mailto:catarina@aguareladigital.com"
              rel="noreferrer"
              target="_blank"
            >
              catarina@aguareladigital.com
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`<ContactBlock /> renders properly in English 1`] = `
<div>
  <section
    class="jsx-590009551 container"
    data-testid="contactBlock_section"
  >
    <div
      class="jsx-590009551 wrapper genericMargins"
    >
      <div
        class="jsx-590009551 titleBlock"
      >
        <strong
          class="jsx-590009551"
        >
          Con
        </strong>
        <strong
          class="jsx-590009551"
        >
          Tac
        </strong>
        <strong
          class="jsx-590009551"
        >
          Tme
        </strong>
      </div>
      <div
        class="jsx-590009551 textBlock"
      >
        <p
          class="jsx-590009551 bodyText bodyText--white"
          data-testid="textShadowBlock_textBlock1"
        >
          Need help with your business, project or brand?
        </p>
        <p
          class="jsx-590009551 bodyText bodyText--white"
        >
          Send me an email and let's talk. We can start as soon as possible defining a social media strategy that meets your objectives.
        </p>
        <p
          class="jsx-590009551 bodyText bodyText--white"
        >
          I am also available to provide social media consulting and share all the tips you need to fulfill your goals.
        </p>
        <div
          class="jsx-2566809609 buttonWrapper"
        >
          <div
            class="jsx-2566809609 buttonBackground"
          >
            <a
              class="jsx-2566809609 button uppercase"
              href="mailto:catarina@aguareladigital.com"
              rel="noreferrer"
              target="_blank"
            >
              catarina@aguareladigital.com
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`<ContactBlock /> renders properly in English and mobile 1`] = `
<div>
  <section
    class="jsx-590009551 container"
    data-testid="contactBlock_section"
  >
    <div
      class="jsx-590009551 wrapper genericMargins"
    >
      <div
        class="jsx-590009551 titleBlock"
      >
        <p
          class="jsx-590009551"
        >
          ConTact
        </p>
      </div>
      <div
        class="jsx-590009551 textBlock"
      >
        <p
          class="jsx-590009551 bodyText bodyText--white"
          data-testid="textShadowBlock_textBlock1"
        >
          Need help with your business, project or brand?
        </p>
        <p
          class="jsx-590009551 bodyText bodyText--white"
        >
          Send me an email and let's talk. We can start as soon as possible defining a social media strategy that meets your objectives.
        </p>
        <p
          class="jsx-590009551 bodyText bodyText--white"
        >
          I am also available to provide social media consulting and share all the tips you need to fulfill your goals.
        </p>
        <div
          class="jsx-2566809609 buttonWrapper"
        >
          <div
            class="jsx-2566809609 buttonBackground alignCenter"
          >
            <a
              class="jsx-2566809609 button uppercase"
              href="mailto:catarina@aguareladigital.com"
              rel="noreferrer"
              target="_blank"
            >
              catarina@aguareladigital.com
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`<ContactBlock /> renders properly on mobile 1`] = `
<div>
  <section
    class="jsx-590009551 container"
    data-testid="contactBlock_section"
  >
    <div
      class="jsx-590009551 wrapper genericMargins"
    >
      <div
        class="jsx-590009551 titleBlock"
      >
        <p
          class="jsx-590009551"
        >
          ConVerSar
        </p>
      </div>
      <div
        class="jsx-590009551 textBlock"
      >
        <p
          class="jsx-590009551 bodyText bodyText--white"
          data-testid="textShadowBlock_textBlock1"
        >
          Precisa de ajuda com o seu negócio, projeto ou marca?
        </p>
        <p
          class="jsx-590009551 bodyText bodyText--white"
        >
          Envie-me um email e vamos conversar. Podemos começar o quanto antes a definir uma estratégia para social media de acordo com os seus objetivos.
        </p>
        <p
          class="jsx-590009551 bodyText bodyText--white"
        >
          Estou ainda disponível para dar consultoria em social media e partilhar todos os passos e dicas de que precisa para cumprir os seus objetivos.
        </p>
        <div
          class="jsx-2566809609 buttonWrapper"
        >
          <div
            class="jsx-2566809609 buttonBackground alignCenter"
          >
            <a
              class="jsx-2566809609 button uppercase"
              href="mailto:catarina@aguareladigital.com"
              rel="noreferrer"
              target="_blank"
            >
              catarina@aguareladigital.com
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
